instance/management/commands/find_orphan_dbs.py

```diff
@@ -0,0 +1,208 @@
+# it under the terms of the GNU Affero General Public License as
+# published by the Free Software Foundation, either version 3 of the
+# License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+"""
+Management command to find orphaned databases
+"""
+import logging
+import traceback
+
+from django.core.management.base import BaseCommand
+import MySQLdb as mysql
+import pymongo
+
+from instance.models.database_server import MongoDBServer, MySQLServer
+from instance.models.openedx_instance import OpenEdXInstance
+
+LOG = logging.getLogger(__name__)
+
+MONGO_SYSTEM_DBS = {"admin", "local", "config"}
+MYSQL_SYSTEM_DBS = {"information_schema", "performance_schema", "sys", "mysql"}
+
+
+class Command(BaseCommand):
+    """
+    find_orphan_dbs management command class
+    """
+
+    help = 'Finds orphaned databases and generates commands to drop them.'
+
+    def add_arguments(self, parser):
+
+        parser.add_argument(
+            "--rm",
+            help="Drop any databases found",
+            action="store_true",
+        )
+
+        subparsers = parser.add_subparsers(
+            title='db type',
+        )
+        mongo_parser = subparsers.add_parser(
+            'mongo', help='Find orphaned MongoDB databases')
+        mongo_parser.add_argument(
+            '--exclude-dbs',
+            help=(
+                'MongoDB databases to ignore when checking for orphans.\n'
+                'System dbs are always excluded.'
+            ),
+            nargs='+',
+            type=str,
+            default=set()
+        )
+        mongo_parser.set_defaults(func=self.find_mongo_orphans)
+
+        mysql_parser = subparsers.add_parser(
+            'mysql', help='Find orphaned MySQL databases')
+        mysql_parser.add_argument(
+            '--exclude-dbs',
+            help=(
+                'MySQL databases to ignore when checking for orphans.\n'
+                'System dbs are always excluded.'
+            ),
+            nargs='+',
+            type=str,
+            default=set()
+        )
+        mysql_parser.set_defaults(func=self.find_mysql_orphans)
+
+    def handle(self, *args, **options):
+        options['func'](**options)
+
+    def find_mongo_orphans(self, exclude_dbs: list, **kwargs) -> int:
+        """
+        Find orphaned databases in MongoDB.
+
+        This searches each MongoDB instance known to OCIM
+        and verifies every database is associated with an
+        OpenEdxInstance.
+        """
+        self.stderr.write("Finding orphaned MongoDB databases...")
```

Comment:
Should this use standard error descriptor? Why not standard output?

Reply:
Historic reasons 🤣 The original intent was to generate output that would be either a script or something that would help to automate the task of dropping the orphaned databases.
I ended up putting the drop feature directly into the task, so it doesn't make much of a difference; however, from a semantics POV, it makes more sense to put these kinds of messages in `stderr`. IIRC, this still puts the list of orphans in `stdout`.